lib/auth/presence/presencev1/service.go

```diff
@@ -41,6 +42,13 @@ type Backend interface {
 	ListRemoteClusters(ctx context.Context, pageSize int, nextToken string) ([]types.RemoteCluster, string, error)
 	UpdateRemoteCluster(ctx context.Context, rc types.RemoteCluster) (types.RemoteCluster, error)
 	PatchRemoteCluster(ctx context.Context, name string, updateFn func(rc types.RemoteCluster) (types.RemoteCluster, error)) (types.RemoteCluster, error)
+
+	UpsertReverseTunnelV2(ctx context.Context, tunnel types.ReverseTunnel) (types.ReverseTunnel, error)
+	DeleteReverseTunnel(ctx context.Context, tunnelName string) error
+}
+
+type Cache interface {
```

Comment:
Soo - when I converted RemoteClusters, I can recall that the GetRemoteClusters/GetRemoteCluster HTTP method called directly thru to the service rather than via the cache - and that this looked somewhat intentional (there was a method on auth.Server that just delegated the call to .Services). I was hesitant to change the behaviour to use the cache as part of the conversion.

More than happy to raise a PR to convert this to use the Cache if you feel that would be safe.

Reply:
Yeah - I had considered that but I wasn't really sure enough if it was intentional or not to codify it as intentional going forward and possibly dissuade someone from switching to the cache in future 😅

Reply:
I think it's for consistency when doing modifications but we should just not modify the `RemoteCluster` object just for advisory purposes IMO, it should be treated as a relatively high importance marker object.

I don't see a lot of functional drawbacks to using the cache, but I don't know if UX would be impacted by not having the cluster in the listings and the web UI immediately after adding a new leaf.